tuf/data/roles.go

```diff
@@ -101,6 +113,111 @@ type Role struct {
 	Email            string   `json:"email,omitempty"`
 }
 
+// BaseRole is an internal representation of a root/targets/snapshot/timestamp role, with its public keys included
+type BaseRole struct {
```

Comment:
Further design work: we're sticking with `BaseRole` and `DelegationRole`, although in our PR we may rename `DelegationRole` to `TargetRole` and the actual targets role will be represented using that with a `Path: [""]` (can sign all paths). That makes walking of all the targets files uniform.